fog/view/enclave/impl/src/lib.rs

```diff
@@ -352,10 +360,31 @@ where
         )
     }
 
+    // Takes each MultiViewStoreResponseQuery
     fn get_minimum_highest_processed_block_count(
-        &self,
-        _shard_query_responses: Vec<(QueryResponse, BlockRange)>,
+        responses: &[DecryptedMultiViewStoreQueryResponse],
     ) -> u64 {
-        todo!()
+        let responses_with_gaps = responses
+            .iter()
+            .cloned()
+            .filter(|response| {
+                response.query_response.highest_processed_block_count
+                    != response.block_range.end_block
+            })
+            .collect::<Vec<DecryptedMultiViewStoreQueryResponse>>();
+
+        let highest_processed_block_count = if !responses_with_gaps.is_empty() {
+            responses_with_gaps
+                .iter()
+                .min_by_key(|response| response.query_response.highest_processed_block_count)
+        } else {
+            responses
+                .iter()
+                .max_by_key(|response| response.query_response.highest_processed_block_count)
+        };
+
+        highest_processed_block_count.map_or(u64::MIN, |response| {
+            response.query_response.highest_processed_block_count
+        })
```

Comment:
:wrench: The cloning and double iteration stood out as a code smell.  I will admit even authoring this suggestion it feels like it might be harder to grok
```suggestion
        let min_block_from_gaps = responses
            .iter()
            .filter(|response| {
                response.query_response.highest_processed_block_count
                    != response.block_range.end_block
            })
            .min_by_key(|response| response.query_response.highest_processed_block_count);

        let highest_processed_block_count = match min_block_from_gaps {
            Some(response) => Some(response),
            _ => responses
                .iter()
                .max_by_key(|response| response.query_response.highest_processed_block_count),
        };

        highest_processed_block_count.map_or(u64::MIN, |response| {
            response.query_response.highest_processed_block_count
        })
```

Reply:
I changed the logic here. So I'll resolve this.